### PR TITLE
[fix](memory): avoid coredump when list pointer is null

### PR DESCRIPTION
**Pull request by levy5307**
# Proposed changes

Issue Number: [5664](https://github.com/apache/doris/issues/5664)

## Problem summary

To avoid the coredump memtioned in above issue, adding the validation of pointer before using it.
BTW, It's a best practise to validate a pointer before using it.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] No
2. Has unit tests been added:
    - [ ] No Need
3. Has document been added or modified:
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/runtime/free_pool.hpp [hunk 1]
```diff
@@ -128,9 +128,14 @@ class FreePool {
 #ifndef NDEBUG
         check_valid_allocation(list);
 #endif
-        // Add node to front of list.
-        node->next = list->next;
-        list->next = node;
+        if (nullptr == list) {
```
Comment (weizuo93) [inline be/src/runtime/free_pool.hpp:131]: Why  is `list` nullptr?

Comment (levy5307) [reply be/src/runtime/free_pool.hpp:131]: > Why is `list` nullptr?

Maybe it was released somewhere

**Comment by weizuo93** [comment]
Normally, the pointer `list` here should not be `nullptr`. I think we'd better to explore the reason why the `list` is nullptr and fix it instead of skipping nullptr, otherwise it might hide the potential problem.

**Comment by xinyiZzz** [comment]
`FreePool` will be removed in the future. simple fixes to try.

**Comment by xinyiZzz** [comment]
@levy5307 p0 case failed

**Pull request closed without merging**